week-14/practices/12-context-practices/practice-0-README.md

```diff
@@ -0,0 +1,95 @@
+# Practice: Context Provider and Consumer Practice
+
+In this practice, you will:
+
+* Set up a Context Provider and Consumer Walkthrough to set the value of a
+context and read it in a component
+
+## Phase 0: Setup
+
+Download the [starter][context-starter] file for this practice.
+
+Open the starter in VSCode and run `npm install` in the terminal.
+
+Run `npm start` to make sure your project is has not run into any errors.
+
+## Phase 1: Create Context
+
+To start, in your `src` directory, create a new directory called `context`. In
+this `context` directory, create a file called `HoroscopeContext.js`. This is
+where all your horoscope context will be placed.
+
+At the top of this file, import `createContext` from `react` and create your
+context called `HoroscopeContext`. You will not need a default value, so leave
+it empty. If you're lost, don't forget to checkout
+[documentation][create-context]. Make sure to `export` it.
+
+Awesome! We have created our context, now let's use our provider.
+
+## Phase 2: Create Context Provider
+
+In your `src/index.js` file, import `HoroscopeContext` from our `context`
+directory.
+
+Inside your `Root` functional component, wrap your `<App/>` with the
+`HoroscopeContext.Provider` component. This is how you will allow your entire
+application access to your horoscope context.
+
+Let's make sure things are going smoothly, so make sure your server is running
+(`npm start` at the root of your starter, which is the same level as your
+`package.json`).
+
+We will now navigate to your browser and open DevTools. If you look at your
+console, you should see a warning message from React that says your
+`Context.Provider` is missing a `value` prop. This means that we have
+successfully created our context provider, but now we must deal with this
+warning message.
+
+## Phase 3: Consume the Context
+
+As you have seen in your DevTools, React is warning us that we need to have a
+`value` prop in our `Context.Provider`. This `value` prop is what holds all
+of our global state. Our goal is to create that pass a value in the `value` prop
+and consume that context.
+
+In the `HoroscopeContext.Provider` component, include your `value` prop and set
+it to an object with the key of `sign` and value of `Leo`. (Feel free to use any
+other sign.) We're expecting the object to look like below:
+
+```javascript
+{"sign": "Leo"}
+```
+
+If you refresh your browser, the warning message should disappear now.
+
+Now it is time to consume the context, so navigate to your `Details` component.
+At the top, import `useContext` from `react` and `HoroscopeContext` from the
+`context` directory. Inside your `Detail` component, invoke `useContext` and
+pass `HoroscopeContext` in as your argument, assign this to a variable.
```

Comment:
**Line 68:** "...assign this to a variable." This could be it's own sentence or a separate idea. It feels weird at the end of the other phrase

Reply:
Does this phrasing sound better? "Inside your `Detail` component, create a variable and have 
it equal to `useContext` invoked with `HoroscopeContext` passed in as your 
argument."